schema/home/home_page_artwork_modules.js

```diff
@@ -1,79 +1,169 @@
 import gravity from '../../lib/loaders/gravity';
 import {
+  filter,
+  find,
+  findIndex,
   keys,
   map,
-  filter,
+  remove,
   slice,
-  findIndex,
   set,
   without,
-  find,
 } from 'lodash';
 import {
-  GraphQLList,
+  GraphQLEnumType,
   GraphQLInt,
+  GraphQLList,
 } from 'graphql';
 import { HomePageArtworkModuleType } from './home_page_artwork_module';
 import loggedOutModules from './logged_out_modules';
 import addGenericGenes from './add_generic_genes';
-import { featuredFair, featuredAuction, relatedArtist } from './fetch';
+import { featuredFair, featuredAuction, relatedArtist, followedGenes } from './fetch';
 
 const filterModules = (modules, max_rails) => {
-  return slice(
-    addGenericGenes(filter(modules, ['display', true])),
-  0, max_rails);
+  const allModules = addGenericGenes(filter(modules, ['display', true]));
+  return max_rails < 0 ? allModules : slice(allModules, 0, max_rails);
 };
 
+const addFollowedGenes = (accessToken, modules, max_followed_gene_rails) => {
+  const followedGeneIndex = findIndex(modules, { key: 'genes' });
+  if (followedGeneIndex && max_followed_gene_rails !== 1) {
+    // 100 is the max that Gravity will return per page.
```

Comment:
Ok, with 250 follows on staging (which is probably slower than production), it takes MP<>Gravity 1.5 sec to perform the `/api/v1/me/follow/genes?size=100` request, which for such an extreme case I don’t find particularly worrying.